File: collect_app/src/main/java/org/odk/collect/android/utilities/ViewUtils.java
package org.odk.collect.android.utilities;

import android.content.Context;

public final class ViewUtils {

    private ViewUtils() {
        
    }

    public static int dpFromPx(final Context context, final float px) {
        return Math.round(px / context.getResources().getDisplayMetrics().density);
    }

    public static int pxFromDp(final Context context, final float dp) {
        return Math.round(dp * context.getResources().getDisplayMetrics().density);
    }

}

File: collect_app/src/main/java/org/odk/collect/android/utilities/ViewUtils.kt
package org.odk.collect.android.utilities

import android.content.Context
import kotlin.math.roundToInt

object ViewUtils {
    @JvmStatic
    fun dpFromPx(context: Context, px: Float): Int {
        return (px / context.resources.displayMetrics.density).roundToInt()
    }

    @JvmStatic
    fun pxFromDp(context: Context, dp: Float): Int {
        return (dp * context.resources.displayMetrics.density).roundToInt()
    }
}
